src/programs/GroupManager.ts

```diff
@@ -101,34 +103,79 @@ export default async function GroupManager(message: Discord.Message, isConfig: b
 
 
 const toggleGroup = async (words:string[], message: Discord.Message) => {
+    if (!isAuthorModerator(message)) {
+        message.react("👎");
+        return;
+    }
+
     words.shift();
     const [ messageId, emoji, channelName ] = words;
     if(!(messageId && emoji && channelName)) {
         message.react("👎")
         message.reply("Invalid syntax, please double check for messageId, emoji, channelName and try again.")
         return;
     }
-    const existingChannel = !!message.guild.channels.cache.find(c => c.name === channelName.toLowerCase());
+    const existingChannel = message.guild.channels.cache.find(c => c.name === channelName.toLowerCase());
     if(!existingChannel) {
         message.react("👎")
         message.reply("That channel doesn't exist here.")
         return;
     }
-    const toggleObject = {
-        messageId,
-        emoji,
-        channelName,
-        originalChannel: message.channel.id
+
+    const reactionMessage = await getOrCreateMessage(messageId);
+
+    if (reactionMessage.channel === null) {
+        message.reply("Since this is the first time I've heard of this message I need your help. " +
+            `Can you put one ${emoji} emoji on the message for me please?\n` +
+            "After you've done that, I'll make sure to put up all the emojis on it. :grin:\n" +
+            "You can keep adding emojis here and add one on the original message when you're done, and I'll add them all!"
+            )
     }
-    if(Tools.updateFile("groupToggle", toggleObject)) {
-        let [ogMessage, channel] = await Tools.getMessageById(messageId, message.guild, message.channel.id)
-        ogMessage = <Discord.Message>ogMessage;
-        ogMessage.react(emoji)
-        message.react("👍")
+
+    const channelToggleRepository = await ChannelToggleRepository();
+
+    try {
+        const toggle = channelToggleRepository.create({
+            emoji,
+            message: reactionMessage,
+            channel: existingChannel.id,
+        });
+        await channelToggleRepository.save(toggle);
+        message.react("👍");
+    } catch (err) {
+        console.error('Failed to create toggle', err)
+        message.react("👎");
+        return
+    }
+
+    if (reactionMessage.channel !== null) {
+        await backfillReactions(reactionMessage.id, reactionMessage.channel, message.guild);
     }
-    
 }
 
+export async function backfillReactions(messageId: string, channelId: string, guild: Guild): Promise<boolean> {
```

Comment:
```suggestion
export async function backfillReactions(messageId: string, channelId: string, guild: Guild) {
```

The return value isn't used anywhere